Immediate.py

```diff
@@ -129,39 +131,57 @@ def parse_imm_type(self, llvm_imm_type: str) -> None:
             )
 
     # RIZIN SPECIFIC
-    def add_code_for_opcode_parsing(self, parsing_code: str) -> None:
-        """Overrides method of parent class. Here we add code which does specific parsing of the operand value on
-        disassembly.
+    @property
+    def c_opcode_parsing(self) -> str | None:
+        """Code which does specific parsing of the operand value on disassembly.
         If the immediate is scaled, has specific attributes or gets extended, the c code for this is added here.
         """
         if self.is_constant:
-            return
+            return None
 
         indent = PluginInfo.LINE_INDENT
```

Comment:
Are those indents necessary? `clang-format` should add them. That's why I prefer to remove them.

Reply:
The indents should indeed not be necessary here.
But conversely, have you considered making the generator emit well-formatted code by itself and skipping clang-format?

Reply:
This was how it was in the beginning. But it leads to a lot of bloated code (think of all the `'{}...'.format(intend * x)` patterns.
Also it is annoying to fix if Githubs CI complains. If something was incorrectly formatted by accident one has to edit it manually (which happened at least to me quite often).
Running clang-format at the end guaranties well formatted code.

Only exceptions are new lines. Those need to be added by us.
clang-format is ok with:
```c
int func_A() {
// Body
}
int func_B() {
// Body
}
```

Whereas we want a newline between functions.

Reply:
Seems reasonable. Then I will strip this explicit formatting from my changes wherever possible and rely on clang-format.

Reply:
removed any added indents